src/myna/application/thesis/melt_pool_geometry_part/execute.py

```diff
@@ -61,20 +71,46 @@ def main():
         proc.wait()
         print(f"\t{pid=}: Simulation complete")
 
+    # Set myna table schema
+    myna_schema = {
+        "x (m)": pl.Float64,
+        "y (m)": pl.Float64,
+        "time (s)": pl.Float64,
+        "length (m)": pl.Float64,
+        "width (m)": pl.Float64,
+        "depth (m)": pl.Float64,
+    }
+
     # Post-process results to convert to Myna format
     if output_files:
-        for mynafile, snapshot_data_file in zip(myna_files, output_files):
-            df = pd.read_csv(snapshot_data_file)
-            df["time (s)"] = df["Time (s)"]
-            df["length (m)"] = df["Length Rotated (m)"]
-            df["width (m)"] = df["Width Rotated (m)"]
-            df["depth (m)"] = df["Depth (m)"]
-            df["x (m)"] = df["Beam X"]
-            df["y (m)"] = df["Beam Y"]
-            df = df[
-                ["x (m)", "y (m)", "time (s)", "length (m)", "width (m)", "depth (m)"]
-            ]
-            df.to_csv(mynafile, index=False)
+        for mynafile, segment_files in zip(myna_files, output_files):
+
+            # Append all segment files
+            thesis_to_myna_mapping = {
```

Comment:
Will this be necessary (eventually) for most any app that works with Pelican? Seems a bit cumbersome, but I don't have a direct suggestion